src/components/MyPage/Edit/Content/index.tsx

```diff
@@ -57,12 +73,44 @@ const Content = ({ profile }: Props) => {
     setProfileImage(url);
   };
 
+  const handleCheckNickname = () => {
+    const originalNickname = profile.member.nickname;
+    const changedNickname = profileState.nickname || '';
+
+    if (originalNickname === changedNickname) {
+      setIsNicknameChecked(false);
+      setIsUsableNickname(true);
+    } else {
+      setIsNicknameChecked(true);
+
+      if (changedNickname.length < 2) {
+        setIsUsableNickname(false);
+        setWarningText('닉네임은 2~15자만 가능합니다.');
+      } else {
+        nicknameCheck(changedNickname).then((response) => {
+          const isDuplicated = response.data.isDuplicated;
+
+          setIsUsableNickname(isDuplicated ? false : true);
+          setWarningText('이미 사용중인 닉네임입니다.');
+        });
+      }
+    }
+  };
+
   const handleCloseSearch = () => {
     setIsMapShow(false);
   };
 
-  const handleSubmitForm = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+  const handleClickSubmit = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     e.preventDefault();
+    const notChangedNickname = profile.member.nickname === profileState.nickname;
+    if (profileState.nickname && profileState.gender && gymState.length) {
+      if (isUsableNickname || notChangedNickname) setIsSubmitModal(true);
+      else setIsNicknameCheckModal(true);
+    } else setIsWarningModal(true);
```

Comment:
여기도 이런 식으로 쓰면 로직을 이해하기 쉬울 것 같다는 생각이 드는데 혹시 어떻게 생각하시나요?

```suggestion
    if (profileState.nickname && profileState.gender && gymState.length) {
      if (isUsableNickname || notChangedNickname) setIsSubmitModal(true);
      setIsNicknameCheckModal(true);
      return;
    } 

    setIsWarningModal(true);
```